app/src/main/java/com/mstudio/superheroarch/presentation/MainActivityPresenter.kt

```diff
@@ -0,0 +1,23 @@
+package com.mstudio.superheroarch.presentation
+
+import com.mstudio.superheroarch.presentation.MainActivityViewModelHelper.changeMainTitle
+import com.mstudio.superheroarch.presentation.MainActivityViewModelHelper.setUpViewModel
+
+class MainActivityPresenter(
+    private val view: MainActivityViewContract
+) {
+
+    fun onCreate() {
+        view.setUpViewModel(setUpViewModel())
+    }
+
+    fun onMainButtonClicked() {
+        view.changeMainTitle(changeMainTitle())
+    }
+
+}
+
+interface MainActivityViewContract {
```

Comment:
You have a title in the MainActivityViewModel, then you have another method to change the title. What's happening with the MainActivityViewModel?